http://bugs.gentoo.org/138653
http://gcc.gnu.org/PR28218

2006-07-04  Paolo Bonzini  <bonzini@gnu.org>

	PR tree-optimization/28218

	* tree-ssa-math-opts.c (execute_cse_reciprocals): Fix calls
	to calculate and free the dominator information.
	    
--- branches/gcc-4_1-branch/gcc/tree-ssa-math-opts.c	2006/07/05 02:15:51	115196
+++ branches/gcc-4_1-branch/gcc/tree-ssa-math-opts.c	2006/07/05 06:20:19	115197
@@ -486,7 +486,8 @@
 				sizeof (struct occurrence),
 				n_basic_blocks / 3 + 1);
 
-  calculate_dominance_info (CDI_DOMINATORS | CDI_POST_DOMINATORS);
+  calculate_dominance_info (CDI_DOMINATORS);
+  calculate_dominance_info (CDI_POST_DOMINATORS);
 
 #ifdef ENABLE_CHECKING
   FOR_EACH_BB (bb)
@@ -523,7 +524,8 @@
 	}
     }
 
-  free_dominance_info (CDI_DOMINATORS | CDI_POST_DOMINATORS);
+  free_dominance_info (CDI_DOMINATORS);
+  free_dominance_info (CDI_POST_DOMINATORS);
   free_alloc_pool (occ_pool);
 }
 
